daphne_worker/src/durable/aggregate_store.rs

```diff
@@ -61,14 +62,16 @@ impl DurableObject for AggregateStore {
             }
 
             (DURABLE_AGGREGATE_STORE_GET, Method::Post) => {
+                // NOTE: The following logic is correct for `max_batch_lifetime = 1`, but
+                // requires changes when we allow batch lifetime longer than 1.
+                let agg_share: DapAggregateShare =
+                    state_get_or_default(&self.state, "agg_share").await?;
                 let collected: bool = state_get_or_default(&self.state, "collected").await?;
                 if !collected {
-                    let agg_share: DapAggregateShare =
-                        state_get_or_default(&self.state, "agg_share").await?;
                     Response::from_json(&AggregateStoreResult::Ok(agg_share))
                 } else {
                     Response::from_json(&AggregateStoreResult::Err(
-                        TransitionFailure::InvalidBatchInterval,
+                        PEER_ERR_BATCH_OVERLAP.to_string(),
```

Comment:
This should be a `DapAbort` variant rather than a string.

Reply:
Since we cannot use DapAbort (we cannot attach `Serialize`, `Deserialize` which is necessary to use it in `from_json`), we defined a new enum `ErrBatchOverlap`.